bodo/utils/utils.py

```diff
@@ -1882,3 +1878,53 @@ class AWSCredentials:
     secret_key: str
     session_token: str | None = None
     region: str | None = None
+
+
+def create_arg_hash(*args, **kwargs):
+    """
+    Create a hash encompassing the string representations of all the args and kwargs.
+    This is typically used to generate a unique and repeatable function name.
+    Args:
+        args and kwargs: the variables that have some effect on the contents of the function.
+    """
+    concat_str_args = "".join(map(str, args)) + "".join(
+        f"{k}={v}" for k, v in kwargs.items()
+    )
+    arg_hash = hashlib.sha256(concat_str_args.encode("utf-8"))
+    return arg_hash.hexdigest()
```

Comment:
Minor question, why do we need this hash method? the builtin `hash` function isn't enough?

Reply:
That is a holdover from the previous code I used for inspiration that I wrote for ramba.  Perhaps it was just paranoia of hash conflicts.  I don't remember the reason I used hashlib rather than standard hash in this use case.  I don't have strong feelings about this one and I can switch to regular hash if you want me to.

Reply:
Not a big deal. Can be follow up PR.